onnxruntime/core/platform/windows/logging/etw_sink.cc

```diff
@@ -160,7 +160,7 @@ void EtwRegistrationManager::LazyInitialize() {
       if (FAILED(etw_status_)) {
         // Registration can fail when running under Low Integrity process, and should be non-fatal
         initialization_status_ = InitializationStatus::Failed;
-        LOGS_DEFAULT(WARNING) << "Error in ETW registration: " << std::to_string(etw_status_);
+        std::cerr << "Error in ETW registration: " << std::to_string(etw_status_) << std::endl;
```

Comment:
are we trying to create this sink as the default logger when creating the environment?

if so it would be good to capture that in a comment so nobody attempts to change it to use LOGS_DEFUALT in the future.

Reply:
We are trying to inject an ETW sink into the logger as an additional sink, on startup if ETW was already listening on process start. This seems both normal, and we want to capture ETW as early as possible. I can add the comment

Reply:
I was aware of some ETW related code in InferenceSession startup (in `InferenceSession::ConstructorCommon`), but the Env should be created prior to that with a default logger and afaik that uses a CLogSink created by `MakePlatformDefaultLogSink` as the default. Is something in the test setup either not creating the Env or explicitly creating the default LoggingManager instance with an ETW based sink?

Reply:
0:000> x
@rax              onnxruntime::logging::LoggingManager::s_default_logger_ = 0x00000000`00000000
0:000> kc10
# Call Site
00 onnxruntime!onnxruntime::logging::LoggingManager::DefaultLogger
01 onnxruntime!onnxruntime::logging::EtwRegistrationManager::LazyInitialize
02 onnxruntime!onnxruntime::logging::EtwRegistrationManager::Instance
03 onnxruntime!onnxruntime::logging::OverrideLevelWithEtw
04 onnxruntime!OrtEnv::GetInstance
05 onnxruntime!OrtApis::CreateEnv
06 onnxruntime_genai!OrtEnv::Create
07 onnxruntime_genai!Generators::OrtGlobals::OrtGlobals
08 onnxruntime_genai!std::make_unique<Generators::OrtGlobals,0>
09 onnxruntime_genai!Generators::GetOrtGlobals
0a onnxruntime_genai!Generators::GetOrtEnv
0b onnxruntime_genai!OgaCreateModelWithRuntimeSettings
0c onnxruntime_genai!OgaCreateModel